src/components/search/SearchBar.jsx

```diff
@@ -27,17 +34,18 @@ const SearchBar = () => {
           return item;
         }
       })
-      .map((item, index) => {
+      .map((item) => {
         return (
-          <Col>
-            <OneCard key={index}>
+          <Col key={item.id} onClick={toggle}>
```

Comment:
Because you're rendering this in a map, it's rendering once for each time map is called. It appears that the last character is rendering, but actually ALL characters are rendering on top of each other, with the last one appearing on top.